src/pack_unpack.c

```diff
@@ -368,7 +383,7 @@ static json_t *pack(scanner_t *s, va_list *ap)
                 prev_token(s);
 
             json = va_arg(*ap, json_t *);
-            if (!json && nullable) {
+            if (s->scan_mode || (!json && nullable)) {
```

Comment:
Could you explain this? Why a JSON null is returned when recovering from an error?

Reply:
When we are already in error recovery mode it is pointless to `json_incref(json)` just so it has to be unreferenced when we finally destroy the object.  It's likely an insignificant optimization to the error path, I can remove it if you want.

Reply:
I think it would be best to remove it, to not be confusing in the future